src/components/cc-oauth-consumer-form/cc-oauth-consumer-form.smart-create.js

```diff
@@ -0,0 +1,95 @@
+// @ts-expect-error FIXME: remove when clever-client exports types
+import { create } from '@clevercloud/client/esm/api/v2/oauth-consumer.js';
+import { snakeCase } from '../../lib/change-case.js';
+import { i18n } from '../../lib/i18n/i18n.js';
+import { notifyError, notifySuccess } from '../../lib/notifications.js';
+import { sendToApi } from '../../lib/send-to-api.js';
+import { defineSmartComponent } from '../../lib/smart/define-smart-component.js';
+import '../cc-smart-container/cc-smart-container.js';
+import { CcOauthConsumerFormDispatchKeyEvent } from './cc-oauth-consumer-form.events.js';
+import './cc-oauth-consumer-form.js';
+
+/**
+ * @typedef {import('./cc-oauth-consumer-form.js').CcOauthConsumerForm} CcOauthConsumerForm
+ * @typedef {import('./cc-oauth-consumer-form.types.js').OauthConsumerWithoutKeyAndSecret} OauthConsumerWithoutKeyAndSecret
+ * @typedef {import('../cc-oauth-consumer-info/cc-oauth-consumer-info.types.js').OauthConsumer} OauthConsumer
+ * @typedef {import('../../lib/send-to-api.types.js').ApiConfig} ApiConfig
+ * @typedef {import('../../lib/smart/smart-component.types.js').OnContextUpdateArgs<CcOauthConsumerForm>} OnContextUpdateArgs
+ */
+
+defineSmartComponent({
+  selector: 'cc-oauth-consumer-form[smart-mode=create]',
+  params: {
+    apiConfig: { type: Object },
+    ownerId: { type: String },
+  },
+  /** @param {OnContextUpdateArgs} args */
+  onContextUpdate({ context, updateComponent, onEvent, component }) {
+    const { apiConfig, ownerId } = context;
+    const api = new Api(apiConfig, ownerId);
+
+    updateComponent('state', { type: 'idle-create' });
+
+    onEvent('cc-oauth-consumer-form-create', (data) => {
+      const oauthConsumerName = data.name;
+      updateComponent('state', (state) => {
+        state.type = 'creating';
+      });
+      api
+        .createOauthConsumer(data)
+        .then(
+          /** @param {OauthConsumer} response */ (response) => {
+            notifySuccess(i18n('cc-oauth-consumer-form.create.success', { oauthConsumerName }));
+            component.resetOauthConsumerForm();
+            updateComponent('state', { type: 'idle-create' });
+            const key = response.key;
+            window.dispatchEvent(new CcOauthConsumerFormDispatchKeyEvent(key));
```

Comment:
fix: I think it's better to dispatch this event from the component instead of window. The event will bubble from component and can be catch more easily

```suggestion
            component.dispatchEvent(new CcOauthConsumerFormDispatchKeyEvent(key));
```